// @shopify/shopify-api/lib/clients/graphql/graphql_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.graphqlClientClass = exports.GraphqlClient = void 0;
var tslib_1 = require("tslib");
var types_1 = require("../../types");
var http_client_1 = require("../http_client/http_client");
var types_2 = require("../http_client/types");
var ShopifyErrors = tslib_1.__importStar(require("../../error"));
var GraphqlClient = /** @class */ (function () {
    function GraphqlClient(params) {
        this.baseApiPath = '/admin/api';
        if (!this.graphqlClass().CONFIG.isPrivateApp &&
            !params.session.accessToken) {
            throw new ShopifyErrors.MissingRequiredArgument('Missing access token when creating GraphQL client');
        }
        this.session = params.session;
        this.client = new (this.graphqlClass().HTTP_CLIENT)({
            domain: this.session.shop,
        });
    }
    GraphqlClient.prototype.query = function (params) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var accessTokenHeader, path, dataType, result;
            var _a;
            return tslib_1.__generator(this, function (_b) {
                switch (_b.label) {
                    case 0:
                        if ((typeof params.data === 'string' && params.data.length === 0) ||
                            Object.entries(params.data).length === 0) {
                            throw new ShopifyErrors.MissingRequiredArgument('Query missing.');
                        }
                        accessTokenHeader = this.getAccessTokenHeader();
                        params.extraHeaders = tslib_1.__assign((_a = {}, _a[accessTokenHeader.header] = accessTokenHeader.value, _a), params.extraHeaders);
                        path = "".concat(this.baseApiPath, "/").concat(this.graphqlClass().CONFIG.apiVersion, "/graphql.json");
                        if (typeof params.data === 'object') {
                            dataType = types_2.DataType.JSON;
                        }
                        else {
                            dataType = types_2.DataType.GraphQL;
                        }
                        return [4 /*yield*/, this.client.post(tslib_1.__assign({ path: path, type: dataType }, params))];
                    case 1:
                        result = _b.sent();
                        if (result.body.errors) {
                            throw new ShopifyErrors.GraphqlQueryError({
                                message: 'GraphQL query returned errors',
                                response: result.body,
                            });
                        }
                        return [2 /*return*/, result];
                }
            });
        });
    };
    GraphqlClient.prototype.getAccessTokenHeader = function () {
        return {
            header: types_1.ShopifyHeader.AccessToken,
            value: this.graphqlClass().CONFIG.isPrivateApp
                ? this.graphqlClass().CONFIG.apiSecretKey
                : this.session.accessToken,
        };
    };
    GraphqlClient.prototype.graphqlClass = function () {
        return this.constructor;
    };
    return GraphqlClient;
}());
exports.GraphqlClient = GraphqlClient;
function graphqlClientClass(params) {
    var config = params.config;
    var HttpClient = params.HttpClient;
    if (!HttpClient) {
        HttpClient = (0, http_client_1.httpClientClass)(params.config);
    }
    var NewGraphqlClient = /** @class */ (function (_super) {
        tslib_1.__extends(NewGraphqlClient, _super);
        function NewGraphqlClient() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        NewGraphqlClient.CONFIG = config;
        NewGraphqlClient.HTTP_CLIENT = HttpClient;
        return NewGraphqlClient;
    }(GraphqlClient));
    Reflect.defineProperty(NewGraphqlClient, 'name', {
        value: 'GraphqlClient',
    });
    return NewGraphqlClient;
}
exports.graphqlClientClass = graphqlClientClass;
