// @shopify/shopify-api/lib/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingReplacementBehavior = exports.BillingInterval = exports.gdprTopics = exports.ClientType = exports.ShopifyHeader = exports.LATEST_API_VERSION = exports.ApiVersion = exports.LogSeverity = void 0;
var LogSeverity;
(function (LogSeverity) {
    LogSeverity[LogSeverity["Error"] = 0] = "Error";
    LogSeverity[LogSeverity["Warning"] = 1] = "Warning";
    LogSeverity[LogSeverity["Info"] = 2] = "Info";
    LogSeverity[LogSeverity["Debug"] = 3] = "Debug";
})(LogSeverity = exports.LogSeverity || (exports.LogSeverity = {}));
var ApiVersion;
(function (ApiVersion) {
    ApiVersion["January22"] = "2022-01";
    ApiVersion["April22"] = "2022-04";
    ApiVersion["July22"] = "2022-07";
    ApiVersion["October22"] = "2022-10";
    ApiVersion["Unstable"] = "unstable";
})(ApiVersion = exports.ApiVersion || (exports.ApiVersion = {}));
exports.LATEST_API_VERSION = ApiVersion.October22;
var ShopifyHeader;
(function (ShopifyHeader) {
    ShopifyHeader["AccessToken"] = "X-Shopify-Access-Token";
    ShopifyHeader["ApiVersion"] = "X-Shopify-API-Version";
    ShopifyHeader["Domain"] = "X-Shopify-Shop-Domain";
    ShopifyHeader["Hmac"] = "X-Shopify-Hmac-Sha256";
    ShopifyHeader["StorefrontAccessToken"] = "X-Shopify-Storefront-Access-Token";
    ShopifyHeader["Topic"] = "X-Shopify-Topic";
    ShopifyHeader["WebhookId"] = "X-Shopify-Webhook-Id";
})(ShopifyHeader = exports.ShopifyHeader || (exports.ShopifyHeader = {}));
var ClientType;
(function (ClientType) {
    ClientType["Rest"] = "rest";
    ClientType["Graphql"] = "graphql";
})(ClientType = exports.ClientType || (exports.ClientType = {}));
exports.gdprTopics = [
    'CUSTOMERS_DATA_REQUEST',
    'CUSTOMERS_REDACT',
    'SHOP_REDACT',
];
var BillingInterval;
(function (BillingInterval) {
    BillingInterval["OneTime"] = "ONE_TIME";
    BillingInterval["Every30Days"] = "EVERY_30_DAYS";
    BillingInterval["Annual"] = "ANNUAL";
    BillingInterval["Usage"] = "USAGE";
})(BillingInterval = exports.BillingInterval || (exports.BillingInterval = {}));
var BillingReplacementBehavior;
(function (BillingReplacementBehavior) {
    BillingReplacementBehavior["ApplyImmediately"] = "APPLY_IMMEDIATELY";
    BillingReplacementBehavior["ApplyOnNextBillingCycle"] = "APPLY_ON_NEXT_BILLING_CYCLE";
    BillingReplacementBehavior["Standard"] = "STANDARD";
})(BillingReplacementBehavior = exports.BillingReplacementBehavior || (exports.BillingReplacementBehavior = {}));
